src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilderTest.java

```diff
@@ -50,39 +48,54 @@
   private static final String TEST_ZONE_B = "us-central1-b";
   private static final String TEST_PROJECT_ID = "test-project-id";
   private static final String OTHER_PROJECT_ID = "other-project-id";
+  private static final String NO_ZONES_PROJECT_ID = "no-zones-project-id";
   private static final String ERROR_PROJECT_ID = "error-project-id";
   private static final String TEST_CLUSTER = "testCluster";
   private static final String OTHER_CLUSTER = "otherCluster";
   private static final String TEST_CREDENTIALS_ID = "test-credentials-id";
-  private static final String EMPTY_PROJECT_CREDENTIALS_ID = "empty-project-credentials-id";
+  private static final String OTHER_CREDENTIALS_ID = "other-credentials-id";
+  private static final String NO_DEFAULT_PROJECT_CREDENTIALS_ID = "empty-project-credentials-id";
   private static final String PROJECT_ERROR_CREDENTIALS_ID = "project-error-credentials-id";
   private static final String ERROR_CREDENTIALS_ID = "error-credentials-id";
   private static final String TEST_ERROR_MESSAGE = "error";
 
-  private static List<Project> listOfProjects;
-  private static List<Zone> listOfZones;
-  private static List<Cluster> listOfClusters;
   private static Jenkins jenkins;
   private static KubernetesEngineBuilder.DescriptorImpl descriptor;
 
   @BeforeClass
   public static void init() throws IOException {
-    listOfProjects = new ArrayList<>();
-    listOfZones = new ArrayList<>();
-    listOfClusters = new ArrayList<>();
     descriptor = Mockito.spy(KubernetesEngineBuilder.DescriptorImpl.class);
     jenkins = Mockito.mock(Jenkins.class);
     ComputeClient computeClient = Mockito.mock(ComputeClient.class);
-    Mockito.when(computeClient.getZones(TEST_PROJECT_ID)).thenReturn(listOfZones);
+    Zone testZoneA = new Zone().setName(TEST_ZONE_A);
```

Comment:
This init method is getting pretty big and seems to incorporate a significant amount of setup that is specific to individual tests. That being the case I think it might be worth considering stripping out the mock setups here and putting them within the unit tests they correspond to. A. that would reduce shared state by having mocks be locally scoped. B. That would make the tests more readable, as they would include their own setup. The downside is that there would be some code redundancy added, but that's usually an acceptable trade-off for testing code in favor of test readability. Also, \@BeforeClass is most useful for operations which are expensive to set up, however mocks are very light weight, so that shouldn't be a consideration.

Reply:
Thank you for this feedback, I think my most recent changes solve the readability concerns with explicitly defined set up before each test, while stubbing out the worst of the set up for the mocks.